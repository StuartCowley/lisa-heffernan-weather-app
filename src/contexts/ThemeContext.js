import React, { createContext, useContext, useState } from "react";
import PropTypes from "prop-types";

const ThemeContext = createContext();

export const useThemeContext = () => useContext(ThemeContext);

export const ThemeContextProvider = ({ children }) => {
  const [useDarkTheme, setUseDarkTheme] = useState(false);
  return (
    <ThemeContext.Provider value={{ useDarkTheme, setUseDarkTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

ThemeContextProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};
