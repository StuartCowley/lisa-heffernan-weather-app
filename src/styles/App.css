.weather-app {
  min-height: 100vh;
}

.weather-app.light {
  background-image: url("../../images/sky-clouds.jpg");
}

.weather-app.dark {
  background-image: url("../../images/dark-sky.jpg");
}