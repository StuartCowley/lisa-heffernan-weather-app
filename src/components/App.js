import React, { useEffect, useState } from "react";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import SearchForm from "./SearchForm";
import ToggleTheme from "./ToggleTheme";
import getForecast from "../requests/getForecast";
import { useThemeContext } from "../contexts/ThemeContext";
import "../styles/App.css";

const App = () => {
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });
  const [selectedDate, setSelectedDate] = useState(0);
  const [searchText, setSearchText] = useState("");
  const [errorMessage, setErrorMessage] = useState("");
  const { useDarkTheme } = useThemeContext();

  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate,
  );

  useEffect(() => {
    getForecast(
      "",
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage,
    );
  }, []);

  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };

  const handleTextSearch = () => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage,
    );
  };

  return (
    <div className={`weather-app ${useDarkTheme ? "dark" : "light"}`}>
      <LocationDetails
        city={location.city}
        country={location.country}
        errorMessage={errorMessage}
      />
      <SearchForm
        searchText={searchText}
        setSearchText={setSearchText}
        onSubmit={handleTextSearch}
      />
      {!errorMessage && (
        <>
          <ForecastSummaries
            forecasts={forecasts}
            onForecastSelect={handleForecastSelect}
            selectedDate={selectedDate}
          />
          {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
        </>
      )}
      <ToggleTheme />
    </div>
  );
};

export default App;
