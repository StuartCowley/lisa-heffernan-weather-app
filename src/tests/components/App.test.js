import React from "react";
import { render } from "@testing-library/react";
import App from "../../components/App";
import { ThemeContextProvider } from "../../contexts/ThemeContext";

describe("App", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <ThemeContextProvider>
        <App />
      </ThemeContextProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
