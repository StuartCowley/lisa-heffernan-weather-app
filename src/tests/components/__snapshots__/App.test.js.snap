// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app light"
  >
    <h1
      class="location-details light"
    />
    <div
      class="search-form light"
    >
      <input
        class="search-form__input"
        type="text"
        value=""
      />
      <button
        class="search-form__button"
        type="submit"
      >
        Search
      </button>
    </div>
    <div
      class="forecast-summaries"
    />
    <div
      class="theme-button"
    >
      <label
        class="switch"
        for="toggle-switch"
      >
        <input
          id="toggle-switch"
          type="checkbox"
        />
        <span
          class="slider round"
        />
      </label>
    </div>
  </div>
</DocumentFragment>
`;
